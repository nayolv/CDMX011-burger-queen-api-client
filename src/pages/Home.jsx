import camarera from "../images/camarera.png";
import Banner from "../components/Banner";
import Cookies from "universal-cookie";
import { useNavigate } from "react-router-dom";


const cookies = new Cookies();
function Home({ loggedUser }) {
  const navigate = useNavigate();

  function desyunos(e) {
    e.preventDefault();
    navigate("./breakfast")
  }
  function comida(e) {
    e.preventDefault();
    navigate("./lunch");
  }
  const userName = cookies.get("name").firstName ;
 
  return (
    <>
    <Banner>
    <p>{userName}<img src={camarera} alt="mesero" /></p>
            </Banner>
      <div className="menu-btns" id="home">
        <button onClick={desyunos}>Desayunos</button>
        <button onClick={comida}>Comida</button>
      </div>
    </>
  );
}

export default Home;
