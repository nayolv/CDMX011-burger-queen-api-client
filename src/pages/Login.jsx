import { Fragment } from "react";
import Banner from "../components/Banner";
import { FormLogin } from "../components/FormLogin";
import camarera from "../images/camarera.png";
import cocinero from "../images/cocinero.png"; 
import gerente from "../images/gerente.png";

function Login() {
  return (
    <Fragment>
      <Banner>
        <p>POS system</p>
      </Banner>
      <main>
        <div className="images">
          <img src={gerente} alt="gerente" />
          <img src={camarera} alt="camarera" />
          <img src={cocinero} alt="cocinero" />
        </div>
        <FormLogin/>
      </main>
    </Fragment>
  );
}
export default Login;
