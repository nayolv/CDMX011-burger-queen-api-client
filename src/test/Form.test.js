import React from "react";
import Form from "../pages/Form";
import Home from "../pages/Home";
import { render, fireEvent } from "@testing-library/react";
import Login from "../pages/Login";


test("Debería renderizar correctamente", () => {
  const component = render(<Form />);
  const email = component.container.querySelector("#email");
  fireEvent.change(email, {
    target: { value: "nattkeres@gmail.com" },
  });
  const password = component.container.querySelector("#password");
  fireEvent.change(password, {
    target: { value: "123456" },
  });
  const button = component.container.querySelector("#login");
  fireEvent.click(button);

  expect(email.value).toBe("nattkeres@gmail.com");
  expect(password.value).toBe("123456");
  expect(render(<Home />)).toMatchSnapshot();
});

test("Debería quedarse en la misma pagina si hay error", () => {
  const component = render(<Form />);
  const email = component.container.querySelector("#email");
  fireEvent.change(email, {
    target: { value: "nattkeres@gmail.com" },
  });
  const password = component.container.querySelector("#password");
  fireEvent.change(password, {
    target: { value: "1234567" },
  });
  const button = component.container.querySelector("#login");
  fireEvent.click(button);

  expect(email.value).toBe("nattkeres@gmail.com");
  expect(password.value).toBe("1234567");
  expect(render(<Login />)).toMatchSnapshot();
});

/*
test('Debería renderizar el Login', () => {
    const component = renderer.create(<Form/>);
    let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Debería devolver error', async () => 
{
let error  = '';
try {
       await signInWithEmailAndPassword(auth, 'example@gmail.com', '1');

} catch (err) {
    error = err.toString()
}
expect(error).toEqual("FirebaseError: Firebase: Error (auth/user-not-found)."
);
});

test('Debería loggear correctamente', async () => {
    const user = await signInWithEmailAndPassword(
        auth,
        'nattkeres@gmail.com',
        '123456'
    );
expect(user.user).toBeTruthy();
});
test('Debería retronar ...', ()=>{

})*/
