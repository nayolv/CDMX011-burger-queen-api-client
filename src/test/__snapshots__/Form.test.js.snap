// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debería quedarse en la misma pagina si hay error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <section
          id="float-father"
        >
          <p>
            Inicia tu sesión
          </p>
          <section
            id="float-label"
          >
            <label
              class="Active"
              for="email"
            >
              Correo Electrónico
            </label>
            <input
              id="email"
              type="text"
              value="nattkeres@gmail.com"
            />
          </section>
          <section
            id="float-label"
          >
            <label
              class="Active"
              for="email"
            >
              Constraseña
            </label>
            <input
              id="password"
              type="password"
              value="1234567"
            />
          </section>
          <button
            id="login"
            type="submit"
          >
            Iniciar sesión
          </button>
        </section>
      </form>
    </div>
    <div>
      <main>
        <div
          class="images"
        >
          <img
            alt="Error al cargar imagen"
            src="gerente.png"
          />
          <img
            alt="Error al cargar imagen"
            src="camarera.png"
          />
          <img
            alt="Error al cargar imagen"
            src="cocinero.png"
          />
        </div>
        <form>
          <section
            id="float-father"
          >
            <p>
              Inicia tu sesión
            </p>
            <section
              id="float-label"
            >
              <label
                class=""
                for="email"
              >
                Correo Electrónico
              </label>
              <input
                id="email"
                type="text"
                value=""
              />
            </section>
            <section
              id="float-label"
            >
              <label
                class=""
                for="email"
              >
                Constraseña
              </label>
              <input
                id="password"
                type="password"
                value=""
              />
            </section>
            <button
              id="login"
              type="submit"
            >
              Iniciar sesión
            </button>
          </section>
        </form>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="images"
      >
        <img
          alt="Error al cargar imagen"
          src="gerente.png"
        />
        <img
          alt="Error al cargar imagen"
          src="camarera.png"
        />
        <img
          alt="Error al cargar imagen"
          src="cocinero.png"
        />
      </div>
      <form>
        <section
          id="float-father"
        >
          <p>
            Inicia tu sesión
          </p>
          <section
            id="float-label"
          >
            <label
              class=""
              for="email"
            >
              Correo Electrónico
            </label>
            <input
              id="email"
              type="text"
              value=""
            />
          </section>
          <section
            id="float-label"
          >
            <label
              class=""
              for="email"
            >
              Constraseña
            </label>
            <input
              id="password"
              type="password"
              value=""
            />
          </section>
          <button
            id="login"
            type="submit"
          >
            Iniciar sesión
          </button>
        </section>
      </form>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Debería renderizar correctamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <section
          id="float-father"
        >
          <p>
            Inicia tu sesión
          </p>
          <section
            id="float-label"
          >
            <label
              class="Active"
              for="email"
            >
              Correo Electrónico
            </label>
            <input
              id="email"
              type="text"
              value="nattkeres@gmail.com"
            />
          </section>
          <section
            id="float-label"
          >
            <label
              class="Active"
              for="email"
            >
              Constraseña
            </label>
            <input
              id="password"
              type="password"
              value="123456"
            />
          </section>
          <button
            id="login"
            type="submit"
          >
            Iniciar sesión
          </button>
        </section>
      </form>
    </div>
    <div>
      <div>
        Holi crayoli soy el Home
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      Holi crayoli soy el Home
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
